lib/bindings/python/rust/http.rs

```diff
@@ -177,8 +177,29 @@ where
     Resp: Data + for<'de> Deserialize<'de>,
 {
     async fn generate(&self, request: SingleIn<Req>) -> Result<ManyOut<Annotated<Resp>>, Error> {
-        match self.0.generate(request).await {
-            Ok(res) => Ok(res),
+        match self.0 .0.generate_in_parts(request).await {
+            Ok((mut stream, context)) => {
+                let request_id = context.id().to_string();
+                let first_item = match futures::StreamExt::next(&mut stream).await {
```

Comment:
we should extract the error message check / check if HTTPError (identical to if let Some(py_err) = e.downcast_ref:)